lib/Basic/Default/TaskQueue.inc

```diff
@@ -100,10 +102,24 @@ bool TaskQueue::execute(TaskBeganCallback Began, TaskFinishedCallback Finished,
         T->Env.empty() ? decltype(Envp)(None)
                        : decltype(Envp)(llvm::toStringRefArray(T->Env.data()));
 
+    llvm::SmallVector<char, 64> stdoutPath;
+    llvm::SmallVector<char, 64> stderrPath;
+    if (fs::createTemporaryFile("stdout", "tmp",  stdoutPath)
+        || fs::createTemporaryFile("stderr", "tmp",  stderrPath)) {
```

Comment:
This is clever, but what deletes the files after the run is finished?

Reply:
Hrm, I think ideally it would register with the compilation via `addTemporaryFile` but that would involve passing the compilation context all the way down which seem unideal. Would just removing the temp files after calling the callback be reasonable? Perhaps also registering with `RemoveFileOnSignal()`?

Reply:
:-/ Maybe? I suppose we don't expect them to be so big (on the offchance we can't clean them up), and they *are* going to be in a system temporary directory.